toqito/state_props/is_separable.py

```diff
@@ -22,14 +20,150 @@
 
+    A quantum state :math:`\rho \in \text{D}(\mathcal{H}_A \otimes \mathcal{H}_B)` is called
+    separable if it can be written as a convex combination of product states. If a state is not
+    separable, it is called entangled.
+
+    Overview
+    ==========
+    This function implements several criteria to determine separability, broadly following a similar
+    order of checks as seen in tools like QETLAB's :code:`IsSeparable` function.
+
+    1.  **Input Validation**: Checks if the input :code:`state` is a square, positive semidefinite (PSD)
+        NumPy array. Normalizes the trace to 1 if necessary. The :code:`dim` parameter specifying
+        subsystem dimensions :math:`d_A, d_B` is processed or inferred.
+
+    2.  **Trivial Cases for Separability**:
+
+        - If either subsystem dimension :math:`d_A` or :math:`d_B` is 1
+          (i.e., :code:`min_dim_val == 1`), the state is always separable.
+
+    3.  **Pure State Check (Schmidt Rank)**:
+
+        - If the input state has rank 1 (i.e., it's a pure state), its Schmidt rank is computed.
+          A pure state is separable if and only if its Schmidt rank is 1 :cite:`WikiScmidtDecomp`.
+        .. note::
+           QETLAB also considers a more general Operator Schmidt Rank condition
+           from :cite:`Cariello_2013_Weak_irreducible` for weak irreducible
+           matrices. This is not explicitly separated in this function but might be
+           covered if such matrices are rank 1. This is a candidate for a
+           future enhancement (see Issue #XYZ).
+
+
+    4.  **Gurvits-Barnum Separable Ball**:
+
+        - Checks if the state lies within the "separable ball" around the
+          maximally mixed state, as defined by Gurvits and Barnum
+          :cite:`Gurvits_2002_Ball`. States within this ball are guaranteed to be
+          separable.
+
+    5.  **PPT Criterion (Peres-Horodecki)**
+        :cite:`Peres_1996_Separability`,
+        :cite:`Horodecki_1996_PPT_small_dimensions`:
+
+        - The Positive Partial Transpose (PPT) criterion is a necessary condition
+          for separability.
+        - If the state is NPT (Not PPT), it is definitively entangled.
+        - If the state is PPT and the total dimension :math:`d_A d_B \le 6`,
+          then PPT is also a *sufficient* condition for separability
+          :cite:`Horodecki_1996_PPT_small_dimensions`.
+
+
+    6.  **3x3 Rank-4 PPT N&S Check (Plücker Coordinates / Breuer / Chen & Djokovic)**:
+
+        - For 3x3 systems, if a PPT state has rank 4, there are known
+          necessary and sufficient conditions for separability. These are often
+          related to the vanishing of the "Chow form" or determinants of
+          matrices constructed from Plücker coordinates of the state's range
+          (e.g., :cite:`Breuer_2006_Optimal`, :cite:`Chen_2013_MultipartiteRank4`).
+          The implementation checks if a specific determinant (:code:`F_det_val`)
```

Comment:
```suggestion
          The implementation checks if a specific determinant
```
Unless you want to define the specific determinant, I would not mention the var in the docstring.